agent/worker.go

```diff
@@ -219,6 +256,39 @@ func reconcileTaskState(ctx context.Context, w *worker, added []*api.Task, remov
 	return tx.Commit()
 }
 
+func reconcileSecrets(ctx context.Context, w *worker, assignments []*api.AssignmentChange, fullSnapshot bool) error {
+	var (
+		updatedSecrets []api.Secret
```

Comment:
Updated the tasks in `reconcileTaskState` to be a list of tasks rather than pointers to tasks.